src/features/Header/components/Header.tsx

```diff
@@ -0,0 +1,119 @@
+import { useHeaderHandlers } from '../hooks/useHeaderHandlers';
+
+import styles from './Header.module.scss';
+
+interface HeaderProps {
+  variant?: 'logo-only' | 'guest' | 'user';
+  user?: UserInfo;
+  onSignin?: () => void;
+  onSignUp?: () => void;
+  onLogout?: () => void;
+  onProfileClick?: () => void;
+  onLogoClick?: () => void;
+  className?: string;
+}
+
+const Header = forwardRef<HTMLElement, HeaderProps>(
+  (
+    {
+      variant = 'guest',
+      user,
+      onSignin: customSignin,
+      onSignUp: customSignUp,
+      onLogout: customLogout,
+      onProfileClick: customProfileClick,
+      onLogoClick: customLogoClick,
+      className,
+    },
+    ref
+  ) => {
+    const defaultHandlers = useHeaderHandlers();
+
+    // 커스텀 핸들러가 있으면 사용, 없으면 기본 핸들러 사용
+    const handlers = {
+      onSignin: customSignin || defaultHandlers.handleSignIn,
+      onSignUp: customSignUp || defaultHandlers.handleSignUp,
+      onLogout: customLogout || defaultHandlers.handleLogout,
+      onProfileClick: customProfileClick || defaultHandlers.handleProfileClick,
+      onLogoClick: customLogoClick || defaultHandlers.handleLogoClick,
+    };
+
+    const renderRightSection = () => {
+      switch (variant) {
+        case 'logo-only':
+          return null;
+
+        case 'guest':
+          return (
+            <div className={styles.rightSection}>
+              <Button
+                size="sm"
+                fontSizePreset="xsmall"
+                fontWeightPreset="regular"
+                className={styles.button}
+                onClick={handlers.onSignUp}
+              >
+                회원가입
+              </Button>
+              <Button
+                size="sm"
+                fontSizePreset="xsmall"
+                fontWeightPreset="regular"
+                className={styles.button}
+                onClick={handlers.onSignin}
+              >
+                로그인
+              </Button>
+            </div>
+          );
+
+        case 'user':
+          return (
+            <div className={styles.rightSection}>
+              <Button
+                size="sm"
+                fontSizePreset="xsmall"
+                fontWeightPreset="regular"
+                className={styles.button}
+                onClick={handlers.onLogout}
+              >
+                로그아웃
+              </Button>
+              {user && (
+                <div className={styles.userProfile} onClick={handlers.onProfileClick}>
+                  <div className={styles.profileImage}>
+                    <img src={user.profileImage} alt={`${user.nickname}의 프로필`} />
+                  </div>
+                  <span className={styles.nickname}>{user.nickname}</span>
+                </div>
+              )}
+            </div>
+          );
+
+        default:
+          return null;
+      }
+    };
+
+    return (
+      <header ref={ref} className={clsx(styles.header, className)} role="banner">
```

Comment:
```suggestion
      <header
        ref={ref}
        className={clsx(styles.header, variant === 'logo-only' && styles.logoOnly, className)}
        role="banner"
      >
```

logo-only 상태일 경우 하단 테두리 존재가 반영되지 않은 것 같습니당
이렇게 수정하시는 건 어떠실까여

Reply:
오 생각도 못하고 있엇다 감사함니다 (줍줍